lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -354,22 +389,39 @@ monitorStakePools tr gp nl db@DBLayer{..} = do
         -> IO (FollowAction ())
     forward blocks (_nodeTip, _pparams) = do
         atomically $ forM_ blocks $ \blk -> do
-            let (slot, registrations) = fromShelleyBlock' getEpochLength blk
-            runExceptT (putPoolProduction (getHeader blk) (getProducer blk)) >>= \case
-                Left e   -> liftIO $ traceWith tr $ MsgErrProduction e
-                Right () -> pure ()
-            forM_ registrations $ \case
-                Registration pool -> do
-                    liftIO $ traceWith tr $ MsgStakePoolRegistration pool
-                    putPoolRegistration slot pool
-                Retirement cert -> do
+            let (slot, certificates) = fromShelleyBlock' getEpochLength blk
+            runExceptT (putPoolProduction (getHeader blk) (getProducer blk))
+                >>= \case
+                    Left e ->
+                        liftIO $ traceWith tr $ MsgErrProduction e
+                    Right () ->
+                        pure ()
+
+            -- A Shelley block can contain multiple certificates relating to
+            -- the same pool.
+            --
+            -- The order in which certificates appear within a block determines
+            -- their precedence relative to one another.
+            --
+            -- Certificates that appear later in a block have higher precedence
+            -- than certificates that appear earlier.
+            --
```

Comment:
```suggestion
            -- When there are multiple certificates for a pool in one block,
            -- certificates that appear later in the block have higher precedence.
            -- It is possible to have a pool (re-)registration and retirement within the same block.
```

Reply:
@rvl I've revised this comment a bit to make it more useful. (See 9efad1d76dc39641e79ed8caeff7f7003493644e)

Let me know if you think it could be revised further.